### ISSUE TITLE
zipstream.unzip, unzipIter, and related functions should all be deprecated

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3666]
|[<img alt="tenth10th's avatar" src="https://avatars.githubusercontent.com/u/854235?s=50" width="50" height="50">](https://github.com/tenth10th)| @tenth10th reported|
|-|-|
|Trac ID|trac#3666|
|Type|defect|
|Created|2009-02-22 22:56:34Z|
|Branch|https://github.com/twisted/twisted/tree/deprecate-zipstream-3666|

The deprecation warning should probably mention that this function is insecure, since you can unzip into parent directories. The code is buggy, and duplicated into unzipIter and unzipIterChunky versions (though, only the Chunky version is really worth keeping, since it doesn't try to load the whole file into memory.)

Attachments:

* [unzip_deprecation.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/6d6/6d6a0e63bc9ef2c1da04c336338d889250d9c24e/08362e5adcfbedef114d5c8e1fa92267a20c1947.diff) (1320 bytes) - added by squiddo on 2011-03-14 02:24:05Z - Deprecation warnings and topfile addition.
* [unzip-deprecation-rev1.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/6d6/6d6a0e63bc9ef2c1da04c336338d889250d9c24e/b7e48c36af2a23befc792eee07e834056a27301a.diff) (3564 bytes) - added by squiddo on 2011-03-14 19:51:53Z - Added tests and suppressed multiple warnings.
* [unzip-deprecation-rev2.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/6d6/6d6a0e63bc9ef2c1da04c336338d889250d9c24e/71dd51c76636da82f879100341c55a652ef11878.diff) (4485 bytes) - added by squiddo on 2011-03-16 23:19:03Z - Removed 'assertRegexpMatches' and suppressed deprecation warnings for deprecated functions.
* [unzip-deprecation-rev3.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/6d6/6d6a0e63bc9ef2c1da04c336338d889250d9c24e/b6622ed793313ee70d96f41e8eed687d13955651.diff) (4478 bytes) - added by squiddo on 2011-03-19 22:23:45Z - Fixed coding standard issues.
* [unzip-deprecation-rev4.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/6d6/6d6a0e63bc9ef2c1da04c336338d889250d9c24e/f1d4cc257fe83b2f2433350c42b2adb0aa48667a.diff) (4476 bytes) - added by lvh on 2011-03-19 22:35:59Z - r3 + whitespace fix supressWarnings = True -> supressWarnings=True in a kwarg

<details><summary>Searchable metadata</summary>

```
trac-id__3666 3666
type__defect defect
reporter__tenth tenth
priority__high high
milestone__ 
branch__branches_deprecate_zipstream_3666 branches/deprecate-zipstream-3666
branch_author__squiddo__lvh squiddo, lvh
status__closed closed
resolution__fixed fixed
component__core core
keywords__security_easy security easy
time__1235343394000000 1235343394000000
changetime__1301096161000000 1301096161000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph set owner to @tenth10th|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

Let's get the ball rolling on this.  I'm about to close [#4802](https://github.com/twisted/twisted/issues/4802) as wontfix because we're just going to get rid of the code in question.

**Comment by twisted-trac** [comment]
|<img alt="squiddo's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|squiddo set owner to squiddo|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="squiddo's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_5"></a>|squiddo commented|
|-|-|

Added deprecation warnings to twisted.python.zipstream.{unzip, unzipIter}. Added topfile. Please review.

**Comment by twisted-trac** [comment]
|<img alt="squiddo's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|squiddo removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)|@lvh set owner to @lvh<br>@lvh set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)<a name="note_8"></a>|@lvh commented|
|-|-|

(In [31019]) Branching to 'deprecate-zipstream-3666'

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)<a name="note_9"></a>|@lvh commented|
|-|-|

I fixed the keywords

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)<a name="note_10"></a>|@lvh commented|
|-|-|

Hey! Thanks for your patch.

It looks good except there need to be tests that these functions are *actually* deprecated.

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)|@lvh set owner to squiddo<br>@lvh set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="squiddo's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|squiddo removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)<a name="note_13"></a>|@lvh set owner to @lvh<br>@lvh set status to `assigned`|
|-|-|

Looks great! Will merge (but kind of backed up at the moment because buildbot is slow :-))

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)<a name="note_14"></a>|@lvh set status to `closed`|
|-|-|

(In [31177]) 
Deprecate twisted.python.zipstream.{unzip,unzipIter}.

Author: squiddo
Reviewer: lvh
Fixes: [#3666](https://github.com/twisted/twisted/issues/3666)

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)<a name="note_15"></a>|@lvh set status to `reopened`|
|-|-|

Derp derp derp don't do things at 2 AM!

assertRegexpMatches is not good because it is new in 2.7. Can you replace it with something else?

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)<a name="note_16"></a>|@lvh commented|
|-|-|

(In [31178]) 
Undo last commit -- stop doing things at 2 AM.

Reopens: [#3666](https://github.com/twisted/twisted/issues/3666)

**Comment by twisted-trac** [comment]
|<img alt="squiddo's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|squiddo set owner to squiddo<br>squiddo set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="squiddo's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_18"></a>|squiddo removed owner|
|-|-|

Now using python 2.5 for twisted testing. Apologies.

Added suppression of warnings in the Zipstream tests for unzip and
unzipIter.

Please review.

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)<a name="note_19"></a>|@lvh commented|
|-|-|

(In [31242]) Branching to 'deprecate-zipstream-3666'

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)<a name="note_20"></a>|@lvh set owner to squiddo|
|-|-|

Twisted coding standard problem: naming. (Sorry about this: it's bitten me more than once myself :()

1. Tests should have a single underscore in them:
   - `test_unzipIter_deprecated` -> `test_unzipIterDeprecated`
   - `test_unzip_deprecated` -> `test_unzipDeprecated`
2. Arguments should be specified in girlyCaps:
   - `suppress_warning` -> `supressWarning` 
3. Default values for a keyword argument should be specified without prefix or suffix space:
   - `suppress_warning = False` -> `suppress_warning=False`

(Also it doesn't apply cleanly to trunk anymore but I can probably apply it manually and figure out why.)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_21"></a>|@exarkun commented|
|-|-|

> Arguments should be specified in girlyCaps

This is really called _camelCase_, _mixedCase_, or _medial capitals_.

**Comment by twisted-trac** [comment]
|<img alt="squiddo's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|squiddo set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)<a name="note_23"></a>|@lvh commented|
|-|-|

Right -- sorry, that's a term I'm used to using internally. mixedCase it is!

**Comment by twisted-trac** [comment]
|<img alt="squiddo's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_24"></a>|squiddo removed owner<br>squiddo set status to `new`|
|-|-|

Fixed coding standard issues.

Please review.

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)<a name="note_25"></a>|@lvh set owner to @lvh|
|-|-|

Missed one on line 256, but I'll fix it for you.

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)<a name="note_26"></a>|@lvh removed owner|
|-|-|

Should pass review, except that some (existing) code violates the 79 chars rule. Perhaps a different ticket?

Please review.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @exarkun<br>@exarkun set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_28"></a>|@exarkun commented|
|-|-|

(In [31379]) Apply unzip-deprecation-rev4.diff 

refs [#3666](https://github.com/twisted/twisted/issues/3666)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_29"></a>|@exarkun commented|
|-|-|

Thanks!

Some minor stuff:

  1. `warning_list` should be named without an underscore
  1. If no assertions are going to be made about the result of `flushWarnings`, `twisted.trial.util.suppress` is generally a better way to go.
  1. `assertEquals` is still preferred over `assertEqual` (at least for now)
  1. Convention is to make an assertion about the warning category and message directly, not just use `assertIn` for keywords.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_30"></a>|@exarkun set status to `closed`|
|-|-|

(In [31433]) Merge deprecate-zipstream-3666

Author: squiddo, lvh
Reviewer: exarkun
Fixes: [#3666](https://github.com/twisted/twisted/issues/3666)

Deprecate `twisted.python.zipstream.unzip` and `twisted.python.zipstream.unzipIter`
in favor of the equivalent stdlib functionality.
